doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -205,9 +423,333 @@ Mail chapel-developers@ for review and merge
+
+.. code-block:: bash
+
+    git pull
+
+    # or:
+    git fetch origin
+    git merge origin/master # replace master with whatever branch you're on
+
+    # similar to:
+    svn update
+
+Update to specific revision number:
+
+.. code-block:: bash
+
+    git checkout <commit sha1>
+
+    # similar to:
+    svn update -r<revision number>
+
+To view "dirty" files, or all those files that are not tracked (includes
+ignored files):
+
+.. code-block:: bash
+
+    git ls-files --others
+
+
+If you've gotten your master branch mucked up but haven't pushed the branch
+with errors to your remote fork, you can fix it with the following series of
+commands:
+
+.. code-block:: bash
+
+   # This will save your old master state to a different branch name, removing
+   # the name "master" from the list of branches you can access on your fork
+   git branch -m <name for old, messed up master>
+
+   # You will get a message indicating you are in a "detached HEAD state".  This
+   # is expected (and desired).  Now the repository you are in is in line with
+   # your fork's master branch.
+   git checkout origin/master
+
+   # This will save the state of the repository right now to a new branch, named
+   # master.
+   git checkout -b master
+
+At this point, a `git push origin master` should work as expected.  Remember, do
+not try this with a master branch that has been corrupted on your remote fork.
+
+An alternate method, if you know or can easily find out the last commit that
+should be kept:
+
+.. code-block:: bash
+
+   # on any branch that contains commits you do not want.
+   git branch <new branch name>
+
+   # do not use --hard if you wish to leave untracked files in your tree
+   git reset --hard <last commit you want to keep>
+
+
+.. _Other logging commands:
+
+Other logging commands
+++++++++++++++++++++++
+
+To view commits grouped by author (for example, show me commits by author from
+1.9.0.1 tag to now):
+
+.. code-block:: bash
+
+    git shortlog --numbered --no-merges
+
+    # With commit sha1 and relative date:
+    git shortlog --numbered --no-merges \
+      --format='* %Cred[%h]%Creset %s %Cgreen(%cr)%Creset'
+
+    # Set alias
+    git config --global alias.sl \
+      'shortlog --numbered --no-merges \
+       --format=\'* %Cred[%h]%Creset %s %Cgreen(%cr)%Creset\''
+
+    # Show commits by author between 1.8.0 and 1.9.0.1 releases:
+    git sl 1.8.0..1.9.0.1
+
+
+Finding a Pull Request by Commit
+++++++++++++++++++++++++++++++++
+
+Suppose you have figured out that a particular commit is causing a problem
+and you'd like to view the pull request discussion on GitHub. You can go
+to
+
+``https://github.com/chapel-lang/chapel/commit/<commit-hash>``
+
+and GitHub shows the pull request number at the bottom of the commit message
+complete with a link to the pull request page.
+
+
+
+.. _Policy details:
+
+Policy details
+~~~~~~~~~~~~~~
+
+.. _Final merge message:
+
+Final merge message
++++++++++++++++++++
```

Comment:
I think I'd like this section to treat both the Final PR message and the Final merge commit message.